server/ServerFSM.cpp

```diff
@@ -2677,125 +2686,29 @@ sc::result WaitingForTurnEndIdle::react(const SaveGameRequest& msg) {
         return discard_event();
     }
 
-    context<WaitingForTurnEnd>().m_save_filename = message.Text();  // store requested save file name in Base state context so that sibling state can retreive it
-
-    return transit<WaitingForSaveData>();
-}
+    const std::string& save_filename = message.Text(); // store requested save file name in Base state context so that sibling state can retreive it
```

Comment:
I think this returns by value, so you shouldn't store it in a reference

Reply:
Fixed.